db/schema.prisma

```diff
@@ -8,26 +8,49 @@ datasource db {
   relationMode = "prisma" // to not worry about foreign ket constrains 
 }
 
+
 model User {
-  id        String    @id @default(cuid())
-  name      String // for react native add @db.Text
-  age       Int @default(0)
-  email     String
+  id            Int       @id @default(autoincrement())
+  username      String @unique
+  balance Int
   createdAt DateTime  @default(now())
-  updatedAt DateTime  @updatedAt
   receipts  Receipt[]
 }
 
 model Receipt {
   id        String   @id @default(cuid()) // UUID (Universally Unique Identifier) or CUID (Collision-resistant Unique Identifier)
   user      User     @relation(fields: [userId], references: [id], onDelete: Cascade) // "onDelete: Cascade" means delteting User will delete all of his receipts 
-  userId    String
-  amount    Int // cents instead of dollars no floating points
+  userId    Int
   createdAt DateTime @default(now())
-  updatedAt DateTime @updatedAt
-
+  products ReceiptProduct[]
   @@index([userId])
 }
 
+model Product {
+   id        String   @id @default(cuid())
+   name String
+   category String 
+   business String 
+   description String
+   tokenValue Int 
+   receipts ReceiptProduct[]
+}
+
+model ReceiptProduct {
```

Comment:
This is one of two ways to do it. The other way was in my pr that got merged. This way links through a seperate table but Prisma has the option to keep track of our relations without creating a foreign key or an extra table